### PR TITLE
support for [TEMPORARY {seconds}] on FT.CREATE

### PR DESCRIPTION
**Pull request by filipecosta90**
- [add] added TEMPORARY index option. Added example ExampleCreateIndex_temporary. 
cc @gkorland

### CODE REVIEW

File: redisearch/schema.go [hunk 1]
```diff
@@ -35,6 +35,42 @@ type Options struct {
...
+	var opts = DefaultOptions
+	return &opts
+}
+
+// If set to true, creates a lightweight temporary index which will expire after the specified period of inactivity.
+// The internal idle timer is reset whenever the index is searched or added to.
+// To enable the temporary index creation, use SetTemporaryPeriod(). This method should be preferably used for disabling the flag
+func (options *Options) SetTemporary(temporary bool) *Options {
+	options.Temporary = temporary
+	return options
```
Comment (MeirShpilraien) [inline redisearch/schema.go:56]: Not sure someone will create a temporary object and then regret ... but it's ok to have this api I guess..

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Go parameters
+GOCMD=go
```
Comment (MeirShpilraien) [inline Makefile:2]: Is it common on go project to have a Makefile?

Comment (filipecosta90) [reply Makefile:2]: I knew someone would pick on the GOCMD =) 
regarding the makefile I believe it helps both for developing, manual testing and CI. cc @rafie

File: redisearch/schema.go (hunk 1 above)
Comment (filipecosta90) [reply redisearch/schema.go:56]: I believe it makes sense if you want to reuse options across several creations. WDYT?

**Pull request merged**